fhir-persistence/src/test/java/com/ibm/fhir/persistence/test/common/AbstractReverseChainTest.java

```diff
@@ -0,0 +1,363 @@
+                                       .build();
+        savedObservation3 = persistence.create(getDefaultPersistenceContext(), savedObservation3).getResource();
+
+        // a Patient that will be referenced by an Observation and references an Organization
+        savedPatient2 = patient.toBuilder().managingOrganization(reference("Organization/" + savedOrg3.getId())).build();
+        savedPatient2 = persistence.create(getDefaultPersistenceContext(), savedPatient2).getResource();
+        savedPatient2 = savedPatient2.toBuilder().name(humanName("Vito", "Corleone")).build();
+        savedPatient2 = persistence.update(getDefaultPersistenceContext(), savedPatient2.getId(), savedPatient2).getResource();
+
+        // an Observation with a reference to a Patient and a reference to an Encounter
+        savedObservation4 = observation.toBuilder()
+                                       .subject(reference("Patient/" + savedPatient2.getId()))
+                                       .encounter(reference("Encounter/" + savedEncounter1.getId()))
+                                       .value(com.ibm.fhir.model.type.String.of("test"))
+                                       .code(CodeableConcept.builder().coding(Coding.builder().code(Code.of("code")).build()).build())
+                                       .build();
+        savedObservation4 = persistence.create(getDefaultPersistenceContext(), savedObservation4).getResource();
+
+        // a Patient that references an Organization and is referenced by an Observation and a Device
+        savedPatient3 = patient.toBuilder().managingOrganization(reference("Organization/" + savedOrg1.getId())).build();
+        savedPatient3 = persistence.create(getDefaultPersistenceContext(), savedPatient3).getResource();
+
+        // an Observation with a reference to a Patient
+        savedObservation5 = observation.toBuilder()
+                                       .subject(reference("Patient/" + savedPatient3.getId()))
+                                       .status(ObservationStatus.FINAL)
+                                       .build();
+        savedObservation5 = persistence.create(getDefaultPersistenceContext(), savedObservation5).getResource();
+
+        // a Device with a reference to a Patient
+        savedDevice1 = device.toBuilder().patient(reference("Patient/" + savedPatient3.getId())).build();
+        savedDevice1 = persistence.create(getDefaultPersistenceContext(), savedDevice1).getResource();
+        savedDevice1 = savedDevice1.toBuilder().manufacturer(string("Updated Manufacturer")).build();
+        savedDevice1 = persistence.update(getDefaultPersistenceContext(), savedDevice1.getId(), savedDevice1).getResource();
+
+        // a Patient that will have no other resources referencing it
+        savedPatient4 = persistence.create(getDefaultPersistenceContext(), patient).getResource();
+    }
+
+    @AfterClass
+    public void deleteResources() throws Exception {
```

Comment:
Random comment, you pick up the style of the code/section very quickly.  Thanks for matching the styles seemingly effortlessly.

Reply:
you're welcome!

Reply:
agreed